pkg/inventory/inventory-client.go

```diff
@@ -507,3 +511,28 @@ func (cic *ClusterInventoryClient) hasSubResource(groupVersion, resource, subres
 	}
 	return false, nil
 }
+
+// getObjStatus returns the list of object status
+// at the beginning of an apply process.
+func getObjStatus(pruneIds, unionIds []object.ObjMetadata) []ObjectStatus {
```

Comment:
Why do we need to generate ObjectStatus in the InventoryClient at all? Shouldn't it always be provided by the caller?

Reply:
This is for the beginning part of the apply process. In the `Merge` function, it computes the union ids and ids to be pruned. So the `ObjectStatus` is not provided by the caller. We need to assign the ObjectStatus for different ids.